[nigelteosw] iP

nigelteosw opened the pull request:
# DukeImproved
> “Your mind is for having ideas, not holding them.” – David Allen ([source](https://dansilvestre.com/productivity-quotes))

DukeImproved frees your mind of having to remember things you need to do. It's,

- text-based
- _easy_ to learn
- ~~FAST~~ SUPER FAST to use
All you need to do is,

1. download it from [here](https://nus-cs2103-ay2223s2.github.io/website/schedule/week4/project.html).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉
5. And it is **FREE**!

Features:

- [x] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)

<hr />

If you Java programmer, you can use it to practice Java too. Here's the main method:
``` java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
```

Commit:
Add Gradle support

nigelteosw committed:
Greet, Echo, Exit

Implement a skeletal version of Duke that starts by greeting the user, simply echos commands entered by the user, and exits when the user types bye.

nigelteosw committed:
Add, List

Add the ability to store whatever text entered by the user and display them back to the user when requested.

nigelteosw committed:
Mark as Done

Add the ability to mark tasks as done. Optionally, add the ability to change the status back to not done.

nigelteosw committed:
ToDos, Events, Deadlines

Add support for tracking three types of tasks:
1. ToDos: tasks without any date/time attached to it e.g., visit new theme park
2. Deadlines: tasks that need to be done before a specific date/time e.g., submit report by 11/10/2019 5pm
3. Events: tasks that start at a specific date/time and ends at a specific date/time
e.g., (a) team project meeting 2/10/2019 2-4pm (b) orientation week 4/10/2019 to 11/10/2019

nigelteosw committed:
Add text UI testing

nigelteosw committed:
Update Duke.java

nigelteosw committed:
Update Duke.java

Add support for deleting tasks from the list.

nigelteosw committed:
Add Storage.java

nigelteosw committed:
Update TaskList.java

nigelteosw committed:
Create duke.txt

nigelteosw committed:
Update duke.txt

nigelteosw committed:
Update duke.txt

nigelteosw committed:
Add LocalDateTime import

nigelteosw committed:
Merge branch 'branch-Level-7'

nigelteosw committed:
Remove duke.txt data

nigelteosw committed:
Merge branch 'branch-Level-8'

nigelteosw committed:
Update Storage.java

nigelteosw committed:
Add More OOP

nigelteosw committed:
Create duke package

nigelteosw committed:
Revert "Create duke package"

This reverts commit c52be2fcc76f9599ba107eb975aea4f44833034b.

nigelteosw committed:
Merge remote-tracking branch 'origin/add-gradle-support'

nigelteosw committed:
Add build.gradle

kayyenl commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,38 @@
+public class Task {
+    protected String description;
+    protected boolean isDone;
+    protected char type;
```
Hey! Kay Yen here. I thought I'd just leave some comments on your code style (its a fast fix, don't worry).

kayyenl commented on src/main/java/TaskList.java:
```diff
@@ -0,0 +1,176 @@
+
+
+public class TaskList {
+    private ArrayList<Task> list;
```
Although this code is just a WIP, I would just like to leave a comment: perhaps changing out the private ArrayList "list" with tasks (note plural), or a more descriptive name would be better! According to the coding conventions, it would be better for a list/arraylist to be named its elements, and in the plural form (with the s). 

> Plural form should be used on names representing a collection of objects. - (the coding document thing)

kayyenl replied:
Overall, I don't see many things (at least to me) that should be changed! I would just like to stress on the collection naming conventions. Add more javadocs and it should be looking good. Thanks so much for reading!

nigelteosw replied:
Thank you for your comments!

kayyenl reviewed (commented):
Just a small fix, overall looking good and mindful of many conventions! 😄

nigelteosw committed:
Add ToDosTest.java

nigelteosw committed:
Update build.gradle

nigelteosw committed:
Rename Task files

nigelteosw committed:
Update TaskList.java

nigelteosw committed:
Update TaskList.java

nigelteosw committed:
Add JavaDoc

nigelteosw committed:
Merge branch 'branch-A-CodingStandard'

nigelteosw committed:
Merge branch 'branch-Level-9'

nigelteosw committed:
Update ToDoTest.java

nigelteosw committed:
implement checkstyle.xml

nigelteosw committed:
Update duke.txt

nigelteosw committed:
Add javaFx part 1-3

nigelteosw committed:
Add javaFx part 4

Enigmatrix commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,38 @@
+
+        ui.showLogo();
+        ui.showLine();
+        Scanner myObj = new Scanner(System.in);
```
perhaps this should be called `scanner`

Enigmatrix commented on src/main/java/duke/Task.java:
```diff
@@ -0,0 +1,47 @@
+public class Task {
+    protected String description;
+    protected boolean isDone;
+    protected char type;
```
this can be an enum

Enigmatrix commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,205 @@
+            }
+            case "D": {
+                LocalDateTime byDate = LocalDateTime.parse(array[3], storageFormatter);
+                this.addTask(new Deadline(array[2], byDate, Boolean.parseBoolean(array[1])));
```
no need to qualify `this`

Enigmatrix commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,205 @@
+     * @param task
+     * @throws DukeException
+     */
+    static void validate(Integer length, Integer error_length, String task) throws DukeException {
```
`error_length` should be `errorLength`

Enigmatrix commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,205 @@
+     * @throws DukeException
+     */
+    public void parser(String userInput) throws DukeException {
+        if (userInput.equals("list")) {
```
overall, this method seems to be quite long. perhaps it can be split into their own command classes

Enigmatrix commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,31 @@
+/**
+ * Represents the UI
+ */
+public class Ui {
```
this class should also take care of the parts where you call `System.out.println` manually, as this is supposed to be a representation of the terminal user interface

nigelteosw committed:
Refactored DialogBox, Launcher, Main...

nigelteosw committed:
Update Duke.java

nigelteosw committed:
Merge branch 'origin/branch-Level-9'

nigelteosw committed:
Update Duke.java

nigelteosw committed:
Merge branch 'origin/branch-A-JavaDoc'

nigelteosw committed:
Create Launcher.java

nigelteosw committed:
Update Deadline.java and Event.java

nigelteosw committed:
Update DialogBox.java

nigelteosw committed:
implement saving when adding tasks

nigelteosw committed:
Update Deadline.java error message

nigelteosw committed:
Merge branch 'heads/origin/branch-A-CodingStandard'

nigelteosw committed:
Update Event.java

nigelteosw committed:
Merge branch 'branch-Level-10'

nigelteosw committed:
Update Duke.java

nigelteosw committed:
Merge branch 'origin/branch-A-CodingStandard'

nigelteosw committed:
Update Duke.java

nigelteosw committed:
Update Duke.java

Duke attributes classes:

Duke may encounter an input that is negative

The value of the index cannot be negative or 0.

Assertions can be used to check if the value that the user has input is 0 or negative.

nigelteosw committed:
Update Event.java

parseCommand: make sure that from date is always before to date

It is impossible for the end date of an event to be before the start date.

Let's,
* add an assertion to check for this

nigelteosw committed:
Clean unused code

nigelteosw committed:
Add PeriodTask.java

nigelteosw committed:
Update Duke.java

nigelteosw committed:
Update Launcher.java

nigelteosw committed:
Update README.md

nigelteosw committed:
Merge pull request #2 from nigelteosw/origin/B-DoWithinPeriodTasks

Update README.md

nigelteosw committed:
Create corky.jpg

nigelteosw committed:
Update MainWindow.java

nigelteosw committed:
Delete DialogBox.java

nigelteosw committed:
Update README.md

nigelteosw committed:
Update README.md

nigelteosw committed:
Update README.md

nigelteosw committed:
Update README.md

nigelteosw committed:
Update README.md

nigelteosw committed:
Add Ui.png

nigelteosw committed:
Update README.md

nigelteosw committed:
Update Main-Class

nigelteosw committed:
Update build.gradle

nigelteosw committed:
Refactoring Codebase

nigelteosw committed:
Add Command.java and other Commands

Followed AB3's implementation of Commands

nigelteosw committed:
Add Response.java

Added a class to handle responses to the user

nigelteosw committed:
Update FindCommand.java

nigelteosw committed:
Update Storage.java

nigelteosw committed:
Update Deadline.java and DeadlineCommand.java

nigelteosw committed:
Update .gitignore

nigelteosw committed:
Update memory.txt

nigelteosw committed:
Update .gitignore

nigelteosw committed:
Update FindCommand.java

nigelteosw committed:
Update EventCommand.java

nigelteosw committed:
Add HelpCommand.java

nigelteosw committed:
Update Duke.java to implement ByeCommand.java

nigelteosw committed:
Update MainWindow.fxml

nigelteosw committed:
Update Storage.java

nigelteosw committed:
Update Storage.java

nigelteosw committed:
Fix Checkstyle errors

nigelteosw committed:
Fix Checkstyle errors

nigelteosw committed:
Fix Checkstyle errors

nigelteosw committed:
Update README.md

nigelteosw committed:
Update README.md

nigelteosw committed:
Update EventCommand.java to remove print statements
